miio/gateway/devices/subdevice.py

```diff
@@ -260,3 +266,67 @@ def get_firmware_version(self) -> Optional[int]:
                 ex,
             )
         return self._fw_ver
+
+    def Register_callback(self, id, callback):
+        """Register a external callback function for updates of this subdevice."""
+        if id in self._registered_callbacks:
+            _LOGGER.error(
+                "A callback with id '%s' was already registed, overwriting previous callback",
+                id,
+            )
+        self._registered_callbacks[id] = callback
+
+    def Remove_callback(self, id):
+        """Remove a external callback using its id."""
+        self._registered_callbacks.pop(id)
+
+    def push_callback(self, action, params):
+        """Push callback received from the push server."""
+        if action not in self.push_events:
+            _LOGGER.error(
+                "Received unregistered action '%s' callback for sid '%s' model '%s'",
+                action,
+                self.sid,
+                self.model,
+            )
+
+        prop = self.push_events[action].get("property")
+        value = self.push_events[action].get("value")
+        if prop is not None and value is not None:
+            self._props[prop] = value
```

Comment:
```suggestion
        event = self.push_events[action]
        prop = event.get("property")
        value = event.get("value")
        if prop is not None:
            self._props[prop] = value
```
I suppose prop&value are optional? If not, this should probably raise an exception or at least warn about invalid event contents?

Reply:
yes prop&value are indeed optional